sql/create.go

```diff
@@ -126,7 +126,28 @@ func (p *planner) CreateTable(n *parser.CreateTable) (planNode, error) {
 	// Inherit permissions from the database descriptor.
 	desc.Privileges = dbDesc.GetPrivileges()
 
-	if err := desc.AllocateIDs(); err != nil {
+	err = desc.AllocateIDs()
+	if err == errMissingPrimaryKey {
```

Comment:
I tried that, and it sadly doesn't work. I duplicated the check in `Validate` for no primary keys, but the fields it relies on to do that are populated near the end of `AllocateIDs`, right before it calls Validate itself. Hence the double call to AllocateIDs in this function.